pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -137,13 +136,20 @@ def build_args(self, test_mode, batch, batch_idx, dataloader_idx):
     def evaluation_step(self, test_mode, batch, batch_idx, dataloader_idx):
         # configure args
         args = self.build_args(test_mode, batch, batch_idx, dataloader_idx)
-
+        has_overfit_batches = (self.trainer.overfit_batches > 0)
         # run actual test step
-        if self.testing:
-            output = self.trainer.accelerator_backend.test_step(args)
-        else:
-            output = self.trainer.accelerator_backend.validation_step(args)
-
+        try:
+            if self.testing:
+                output = self.trainer.accelerator_backend.test_step(args)
+            else:
+                output = self.trainer.accelerator_backend.validation_step(args)
+        except TypeError:
```

Comment:
what if the typeerror is coming from some other bug? it will make it almost impossible to debug this because an error will occur later, unrelated to this part of the code here. can we not just check if step takes multiple dataloaders? 
related: #2266

Reply:
We only have access to `dataloader_idx`, `val_dataloader` and `test_dataloader` became `train_dataloader` if there is `overfit_batches`